lib/wda/iproxy.js

```diff
@@ -1,98 +1,33 @@
-import log from '../logger';
-import { killProcess } from './utils';
-import B from 'bluebird';
-import { logger } from 'appium-support';
-import { SubProcess } from 'teen_process';
-import { checkPortStatus } from 'portscanner';
-import { waitForCondition } from 'asyncbox';
-
-
-const IPROXY_STARTUP_TIMEOUT = 5000;
-
-const iproxyLog = logger.getLogger('iProxy');
+import { utilities } from 'appium-ios-device';
+import net from 'net';
 
 class iProxy {
-  constructor (udid, localport, deviceport, detached = true) {
-    this.expectIProxyErrors = true;
+  constructor (udid, localport, deviceport) {
     this.localport = parseInt(localport, 10);
     this.deviceport = parseInt(deviceport, 10);
     this.udid = udid;
-    this.iproxy = new SubProcess('iproxy', [localport, deviceport, udid], {
-      detached,
-      stdio: ['ignore', 'pipe', 'pipe'],
-    });
   }
 
-  async start () {
-    log.debug(`Starting iproxy to forward traffic from local port ${this.localport} ` +
-      `to device port ${this.deviceport} over USB for the device ${this.udid}`);
-    this.expectIProxyErrors = true;
-
-    return await new B((resolve, reject) => {
-      this.iproxy.on('exit', (code) => {
-        log.debug(`iproxy exited with code '${code}'`);
-        if (code) {
-          return reject(new Error(`iproxy exited with code '${code}'`));
-        }
-      });
-      this.iproxy.on('output', (stdout, stderr) => {
-        // do nothing if we expect errors
-        if (this.expectIProxyErrors) {
-          return;
-        }
-
-        let out = stdout || stderr;
-        for (let line of out.split('\n')) {
-          if (!line.length) {
-            continue;
-          }
-
-          if (line.includes('Resource temporarily unavailable')) {
-            // this generally happens when WDA does not respond,
-            // so print a more useful message
-            log.debug('Connection to WDA timed out');
-          } else {
-            iproxyLog.debug(line);
-          }
-        }
-      });
+  start () {
```

Comment:
is it guaranteed that after `start` invocation is completed the proxy is ready to accept commands?

Reply:
do we care if start is invoked twice in a row?

Reply:
No, it will fail to connect like `iproxy` until WDA is up

Reply:
I didn't care about that and hoped that it wouldn't be called multiple times

Reply:
I implemented a safeguard to prevent calling start multiple times